### PR TITLE
Integrate automatic period determination into AutoMLSearch

### LINKED ISSUE #3985: Integrate Automatic Periodicity Determination into EvalML

### PR DESCRIPTION
**Pull request by eccabay**
Resolves #3985

### CODE REVIEW

File: evalml/pipelines/utils.py [hunk 1]
```diff
@@ -250,7 +250,12 @@ def _get_decomposer(X, y, problem_type, estimator_class, sampler_name=None):
             return components
         freq = time_index.freq.name
         if STLDecomposer.is_freq_valid(freq):
-            components.append(STLDecomposer)
+            # Make sure there's a seasonal period
+            order = 3 if "Q" in freq else 5
+            temp_decomp = STLDecomposer(time_index, rel_max_order=order)
+            seasonal_period = temp_decomp.determine_periodicity(X, y)
```
Comment (jeremyliweishih) [inline evalml/pipelines/utils.py:256]: if we're utilizing `determine_periodicity` like this - maybe it should be a class method?

File: evalml/tests/component_tests/decomposer_tests/test_stl_decomposer.py [hunk 2]
```diff
@@ -38,39 +38,57 @@ def test_stl_raises_warning_high_smoother(caplog, ts_data):
...
+):
+
+    X, y = generate_seasonal_data(real_or_synthetic="synthetic")(
+        period,
+        freq_str=freq,
+    )
+
+    stl = STLDecomposer()
+    stl.fit(X, y)
+    assert period * 0.99 <= stl.period <= period * 1.01
```
Comment (jeremyliweishih) [inline evalml/tests/component_tests/decomposer_tests/test_stl_decomposer.py:64]: can you add a comment why we use `0.99` and `1.01`? Looks somewhat arbitrary unless you have the context!

**Review by jeremyliweishih — APPROVED** [review]
Great work! Just some nits 😄

File: evalml/pipelines/components/transformers/preprocessing/stl_decomposer.py [hunk 3]
```diff
@@ -36,24 +36,7 @@ class STLDecomposer(Decomposer):
...
-        "L",
-        "U",
-        "N",
-        "A",
-        "BA",
-        "AS",
-        "BAS",
-        "BH",
-    ]
+    invalid_frequencies = []
```
Comment (christopherbunn) [inline evalml/pipelines/components/transformers/preprocessing/stl_decomposer.py:39]: Can we delete this line entirely if there are no invalid frequencies?

File: evalml/tests/component_tests/decomposer_tests/test_stl_decomposer.py (hunk 2 above)
Comment (christopherbunn) [reply evalml/tests/component_tests/decomposer_tests/test_stl_decomposer.py:64]: I'm assuming there's a margin of error with the period determination that this test accounts for?

**Review by christopherbunn — APPROVED** [review]
LGTM once Jeremy's comments are addressed as well!

File: evalml/tests/component_tests/decomposer_tests/test_stl_decomposer.py [hunk 4]
```diff
@@ -294,33 +311,22 @@ def test_stl_decomposer_get_trend_dataframe_sets_time_index_internally(
...
-        ),
-    ],
+    ["T", "A"],
 )
 def test_unsupported_frequencies(
     bad_frequency,
     generate_seasonal_data,
 ):
-    """This test exists to highlight that the underlying statsmodels STL component won't work for minute or annual frequencies."""
+    """This test exists to highlight that even though the underlying statsmodels STL component won't work
```
Comment (chukarsten) [inline evalml/tests/component_tests/decomposer_tests/test_stl_decomposer.py:320]: 👍

**Review by chukarsten — APPROVED** [review]
Looks good!  Well done.  Nice type hinting!

**Comment by machineFL** [comment]
No linked issues found. Please add the corresponding issues in the pull request description. <br/> [Use GitHub automation to close the issue when a PR is merged](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)

**Pull request merged**